File: 6 - Python/sentimental/readability/readability.py
from math import ceil, floor

def main():
    text = input("Text: ")

    w = words(text)

    l = (letters(text)/w) * 100
    s = (sentences(text)/w) * 100

    grade = 0.0588 * l - 0.296 * s - 15.8

    if grade%1 < 0.5:
        grade = floor(grade)
    else:
        grade = ceil(grade)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def letters(string):
    countletters = 0

    for i in string:
        if i.lower() in "abcdefghijklmnopqrstuvwxyz":
            countletters += 1

    return countletters


def words(string):
    return string.count(' ') + 1


def sentences(string):
    return string.count('.') + string.count('!') + string.count('?')


main()

File: 2 - Arrays/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int letters(string l);
int words(string w);
int sentences(string s);
int approximate(float any);

int main(void){

    string text = get_string("Text: ");

    int ql = letters(text);
    int qw = words(text);
    int qs = sentences(text);

    float L = ((float) ql/qw) * 100;
    float S = ((float) qs/qw) * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    index = approximate(index);

    if(index < 1) printf("Before Grade 1\n");
    else if(index > 16) printf("Grade 16+\n");
    else printf("Grade %i\n", (int) index);

}

int letters(string l){

    int total = 0;

    for(int i = 0, len = strlen(l); i < len; i++){
        if((l[i] >= 'A' || l[i] >= 'a') && (l[i] <= 'Z' || l[i] <= 'z')) total++;
    }

    return total;
}

int words(string w){
    int total = 1;

    for(int i = 0, len = strlen(w); i < len; i++){
        if(w[i] == ' ') total++;
    }

    return total;
}

int sentences(string s){
    int total = 0;

    for(int i = 0, len = strlen(s); i < len; i++){
        if(s[i] == '?' || s[i] == '.' || s[i] == '!') total++;
    }

    return total;
}

int approximate(float any){

    any *= 1000;

    int number = any/1000;

    if ( (int) any%1000 > 500) number++;

    return number;
}
